bulk_add catch exception to not discard all data

doomedraven opened the pull request:
the uniq thing which we should fix after this is unicode url regex support, i will add an issue to keep track of that

tomchop commented on core/web/api/investigation.py:
```diff
@@ -99,14 +100,18 @@ def bulk_add(self, id):
+                    if node['new_tags']:
+                        n.tag(node['new_tags'].split(', '))
+                    nodes.append(n)
+                except Exception as e:
```
It would be good to catch a more specific type of exception here. Are you looking for `ObservableValidationError`?

doomedraven replied:
thanks for point for better solution, updated

doomedraven commented:
Oh thanks for the point, will update once i will be at home

El sáb., 11 may. 2019 18:07, Thomas Chopitea <notifications@github.com>
escribió:

> *@tomchop* requested changes on this pull request.
> ------------------------------
>
> In core/web/api/investigation.py
> <https://github.com/yeti-platform/yeti/pull/397#discussion_r283102354>:
>
> > -                    _type = globals()[node['type']]
> -
> -                n = _type.get_or_create(value=node['value'])
> -                if node['new_tags']:
> -                    n.tag(node['new_tags'].split(', '))
> -                nodes.append(n)
> +                try:
> +                    if node['type'] in globals() and issubclass(
> +                            globals()[node['type']], Observable):
> +                        _type = globals()[node['type']]
> +
> +                    n = _type.get_or_create(value=node['value'])
> +                    if node['new_tags']:
> +                        n.tag(node['new_tags'].split(', '))
> +                    nodes.append(n)
> +                except Exception as e:
>
> It would be good to catch a more specific type of exception here. Are you
> looking for ObservableValidationError?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeti-platform/yeti/pull/397#pullrequestreview-236383119>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOFH333A4ICGPM2WTMZLUTPU3VLTANCNFSM4HLRV6WQ>
> .
>

doomedraven commented:
should be fine now no=
